src/snowflake/cli/_plugins/nativeapp/v2_conversions/v2_to_v1_decorator.py

```diff
@@ -48,28 +51,37 @@ def _convert_v2_artifact_to_v1_dict(
     return v2_artifact
 
 
-def _pdf_v2_to_v1(v2_definition: DefinitionV20) -> DefinitionV11:
+def _pdf_v2_to_v1(
+    v2_definition: DefinitionV20,
+    package_entity_id: str = "",
+    app_entity_id: str = "",
+) -> DefinitionV11:
     pdfv1: Dict[str, Any] = {"definition_version": "1.1", "native_app": {}}
 
     app_package_definition: Optional[ApplicationPackageEntityModel] = None
     app_definition: Optional[ApplicationEntityModel] = None
 
     for key, entity in v2_definition.entities.items():
         if entity.get_type() == ApplicationPackageEntityModel.get_type():
-            if app_package_definition:
-                raise ClickException(
-                    "More than one application package entity exists in the project definition file."
-                )
-            app_package_definition = entity
+            if entity.entity_id == package_entity_id or not package_entity_id:
+                if app_package_definition:
+                    raise ClickException(
+                        "More than one application package entity exists in the project definition file, "
+                        "specify --package-entity-id to choose which one to operate on."
+                    )
+                app_package_definition = entity
```

Comment:
out of curiosity - if pdf has two package entities with the same id, do we fail at a step before this? 
same question for any entity type really. where do we fail the case of two entities with the same type and id?

Reply:
you can't have two entities with the same ID since they're stored in a map/dict and the ID is the key